File: src/application/decorator/strategies/clear_strategy.py
from collections.abc import Sequence
from application.decorator.strategies.abstracts.base_helper_strategy import BaseHelperStrategy
from application.decorator.strategies.decorator.decorator_helper_strategy import strategy
from application.enums.user_action import UserAction

@strategy(priority=10)
class ClearStrategy(BaseHelperStrategy):
    _PHRASES = ["cls", "clear"]

    def can_handle(self, normalized: str, tokens: Sequence[str]) -> bool:
        return normalized.replace(" ", "") in self._PHRASES

    def handle(self, normalized: str, tokens: Sequence[str]):
        return UserAction.CLEAR, None

File: angular-app/src/app/core/application/decorator/helpers/strategies/clear-strategy.ts
import { UserAction } from '../../../enums/user-action';
import { BaseHelperStrategy } from './abstracts/base-helper-strategy';

const phrases = ['cls', 'clear'];

export class ClearStrategy extends BaseHelperStrategy {

  override priority = 10;

  override canHandle(normalized: string, tokens: string[]): boolean {
    return phrases.some(p => normalized.replace(/ /g, '') === p.replace(/ /g, ''));
  }

  override handle(normalized: string, tokens: string[]): [UserAction, null] {
    return [UserAction.CLEAR, null];
  }
}
